opencti-platform/opencti-graphql/src/database/middleware.js

```diff
@@ -2380,6 +2380,10 @@ const upsertElement = async (context, user, element, type, basePatch, opts = {})
   if (type === ENTITY_TYPE_INDICATOR) {
     if (updatePatch.decay_applied_rule && updatePatch.decay_base_score === element.decay_base_score) {
       logApp.debug('UPSERT INDICATOR -- no decay reset because no score change', { element, basePatch });
+      updatePatch.x_opencti_score = element.x_opencti_score; // don't change the score
+      // don't reset valid_from & valid_until
+      updatePatch.valid_from = element.valid_from;
```

Comment:
I'm not really sure about this part, I wonder if we might have a case where valid_from / valid_until change with the same score